src/components/Checkbox/Checkbox.spec.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {Checkbox} from './index';
+
+describe('Checkbox', () => {
+    it('should display additional class names', () => {
+        const className = 'test';
+        render(<Checkbox aria-label="checkbox" className={className}/>);
+        // Checking the parent element because additional classes are applied to the
+        // wrapping div whereas custom attributes are applied to the inner input
+        // element
+        expect(screen.getByRole('checkbox').parentElement).toHaveClass(className);
+    });
+
+    it('should add additional attributes', () => {
+        const customAttribute = 'test';
+        render(<Checkbox aria-label="checkbox" data-custom={customAttribute}/>);
+        expect(screen.getByRole('checkbox')).toHaveAttribute('data-custom', customAttribute);
+    });
+
+    it('should check off when clicked on', () => {
+        render(<Checkbox aria-label="checkbox"/>);
+        const checkbox = screen.getByRole('checkbox');
+        userEvent.click(checkbox);
+        expect(checkbox).toBeChecked();
+    });
+
+    it('should un-check when clicked on twice', () => {
+        render(<Checkbox aria-label="checkbox"/>);
+        const checkbox = screen.getByRole('checkbox');
+        userEvent.click(checkbox);
+        userEvent.click(checkbox);
+        expect(checkbox).not.toBeChecked();
+    });
+
+    it('should initially be checked off when the defaultSelected prop is set', () => {
+        render(<Checkbox defaultSelected aria-label="checkbox"/>);
+        expect(screen.getByRole('checkbox')).toBeChecked();
+    });
+
+    it('should have mixed state when specified with the isIndeterminate prop', () => {
+        render(<Checkbox isIndeterminate aria-label="checkbox"/>);
+        expect(screen.getByRole('checkbox')).toBePartiallyChecked();
+    });
```

Comment:
As these events are managed by `react-aria` do we need to add unit tests for them?

Reply:
Hmm, these props are created by `react-aria`, however, we are implementing the hooks they provide into a component. I'm kind of on the fence on this one 🤔